jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/SpansToDependencyLinks.java

```diff
@@ -143,14 +142,14 @@ static boolean isServerSpan(Span span) {
         return dependencies;
     }
 
-    private Optional<Dependency> sharedSpanDependency(Set<Span> sharedSpans) {
+    protected Optional<Dependency> sharedSpanDependency(Set<Span> sharedSpans) {
         String clientService = null;
         String serverService = null;
         for (Span span: sharedSpans) {
             for (KeyValue tag: span.getTags()) {
-                if (Tags.SPAN_KIND_CLIENT.equals(tag.getValueString())) {
+                if (Tags.SPAN_KIND_CLIENT.equals(tag.getValueString()) || Tags.SPAN_KIND_CONSUMER.equals(tag.getValueString())) {
```

Comment:
Shouldn't these be vice-versa? Producer for client and consumer for server?

Reply:
As i get it:
Client - child, for example http service, is the one who receiving requests
Consumer in event buses the one who listen for messages(client) that producer - publish(server)

Reply:
![222 0](https://user-images.githubusercontent.com/11577813/65669794-d956ee00-e04c-11e9-917c-5e52a4b4c504.png)

Reply:
Client span models HTTP client incocation. E.g. curl call to server.
Server span models processing of the request on the server-side.

Given the above producer is similar to a client and consumer to the server. 

IIRC @objectiser has pulled this into the spec. He can confirm which one is correct.

Reply:
@pavolloffay Ohh yea, you are right, i'll swap it after coming home